encodings/runend/src/compute/filter.rs

```diff
@@ -0,0 +1,150 @@
+use std::cmp::min;
+use std::ops::AddAssign;
+
+use arrow_buffer::BooleanBuffer;
+use num_traits::AsPrimitive;
+use vortex_array::array::PrimitiveArray;
+use vortex_array::compute::{filter, FilterFn, FilterMask};
+use vortex_array::validity::Validity;
+use vortex_array::variants::PrimitiveArrayTrait;
+use vortex_array::{ArrayData, ArrayLen, IntoArrayData, IntoArrayVariant};
+use vortex_buffer::buffer_mut;
+use vortex_dtype::{match_each_unsigned_integer_ptype, NativePType};
+use vortex_error::{VortexResult, VortexUnwrap};
+
+use crate::compute::take::take_indices_unchecked;
+use crate::{RunEndArray, RunEndEncoding};
+
+const FILTER_TAKE_THRESHOLD: f64 = 0.001;
+
+impl FilterFn<RunEndArray> for RunEndEncoding {
+    fn filter(&self, array: &RunEndArray, mask: &FilterMask) -> VortexResult<ArrayData> {
+        // If we are taking less than 0.1% of values use take
```

Comment:
Surely this should be based on a true_count, rather than a ratio?

Reply:
Or maybe a `true_count / ends.len()` ?

Reply:
Probably the latter, you’re right we want to penalize taking from a run

Reply:
funny enough `true_count / ends.len()` < 0.001 is too restrictive and we end up not taking the branch

Reply:
Well yeah, you'd have to change the 0.001! Also.. you definitely converted to floats before dividing?

I think we need a benchmark for this tbh, don't want to just merge a random number

Reply:
I have, ok I will close it for now and will reopen when I have time to write a benchmark